### PR TITLE
[MRG] doccheck 2: preparation

### PR DESCRIPTION
**Pull request by banilo**

### CODE REVIEW

File: doc/manipulating_visualizing/data_preparation.rst [hunk 1]
```diff
@@ -12,31 +12,40 @@ Data preparation: loading and basic transformation
 
 .. topic:: **File names as arguments**
 
-   For most functions or objects, it is not necessary to explicitely load
-   the data. Indeed, most of nilearn functions can take file names as
+   For most applications of nilearn, it is not necessary to load
+   the neuroimaging data oneself.
+   Rather raw data (i.e., numpy arrays),
```
Comment (AlexandreAbraham) [inline doc/manipulating_visualizing/data_preparation.rst]: Rather than?

File: doc/manipulating_visualizing/data_preparation.rst [hunk 2]
```diff
@@ -78,15 +89,15 @@ have to call low-level functions (see e.g. :mod:`nilearn.signal`,
 :class:`NiftiMasker`: loading, masking and filtering
 =========================================================
 
-This section describes how to use the :class:`NiftiMasker` class in
-more details than the previous description. :class:`NiftiMasker` is a
+This section details how to use the :class:`NiftiMasker` class
```
Comment (AlexandreAbraham) [inline doc/manipulating_visualizing/data_preparation.rst]: MIssing period at the end of the sentence?

File: doc/manipulating_visualizing/data_preparation.rst [hunk 3]
```diff
@@ -112,10 +123,12 @@ in memory:
...
 
-In the basic tutorial, we showed how the masker could compute a mask
-automatically, and it has done a good job. But, on some datasets, the
-default algorithm performs poorly. This is why it is very important to
-**always look at how your data look like**.
+In this tutorial, we show how the masker object can compute a mask
+automatically for subsequent statistical analysis.
+On some datasets, the default algorithm may however perform poorly.
+This is why it is very important to
+**always look at look at your data look** before and after feature
```
Comment (AlexandreAbraham) [inline doc/manipulating_visualizing/data_preparation.rst]: Look, maybe there are too much look in this look.

File: doc/manipulating_visualizing/data_preparation.rst [hunk 4]
```diff
@@ -185,13 +204,14 @@ voxels that are very light on the EPI image.
 
 
 
-Common data preparation steps: resampling, smoothing, filtering
-----------------------------------------------------------------
+Common data preprocessing steps: resampling, smoothing, filtering
```
Comment (AlexandreAbraham) [inline doc/manipulating_visualizing/data_preparation.rst]: We wanted to avoid the word `preprocessing` that is specific to operations applied before what we do here.

File: doc/manipulating_visualizing/data_preparation.rst [hunk 5]
```diff
@@ -199,21 +219,28 @@ Common data preparation steps: resampling, smoothing, filtering
...
-lowering image resolution. Certain image viewers also require images to be
+       affine (i.e., matrix transforming from voxel space into world space).
+       Additionally, a *target_shape* can be used to resize
+       images (i.e., cropping or padding with zeros) to match an
+       expected data image dimensions (shape composed of x, y, and z).
+
+As a common use case, resampling can be a viable means to
+downsample image quality on purpose to increase processing speed
+and lower memory consumption of an analysis pipeline.
+In fact, certain image viewers also require images to be
```
Comment (AlexandreAbraham) [inline doc/manipulating_visualizing/data_preparation.rst]: I think we can explicitely cite fslview.

File: doc/manipulating_visualizing/data_preparation.rst [hunk 6]
```diff
@@ -244,11 +271,14 @@ computes automatically the translation part of the affine.
 Smoothing
 .........
 
-If smoothing the data prior to converting to voxel signals is required, it can
-be performed by :class:`NiftiMasker`. It is achieved by passing the full-width
-half maximum (in millimeter) along each axis in the parameter `smoothing_fwhm`.
+:class:`NiftiMasker` can further be used for local spatial filtering of
+the neuroimaging data to make the data more homogeneous and thus account
+for inter-individual differences in neuroanatomy and preprocessing.
```
Comment (AlexandreAbraham) [inline doc/manipulating_visualizing/data_preparation.rst]: Referring to my previous note. If you use `preprocessing` here, people may be troubled because you call smoothing itself `preprocessing`.

File: doc/manipulating_visualizing/data_preparation.rst [hunk 7]
```diff
@@ -310,51 +350,41 @@ Extraction of signals from regions:\  :class:`NiftiLabelsMasker`, :class:`NiftiM
...
-  represented (no weighting).
+  common label (e.g., anatomical brain region definitions as integers)
+  in the region definition array. The set of
+  regions is defined by a single 3D array, containing a voxel-wise
+  dictionary of label numbers that denote what
+  region a given voxel belongs to. This technique has a big advantage: the
+  required memory load is independent of the number of regions, allowing
+  for a large number of regions. On the other hand, there are
+  several disadvantages: regions cannot topographically overlap
+  (i.e., regions cannot overlap)
```
Comment (AlexandreAbraham) [inline doc/manipulating_visualizing/data_preparation.rst]: I don't know if the addition of `topographically` will make sense to most of the readers.

**Comment by AlexandreAbraham** [comment]
Great improvements, as usual!

File: doc/manipulating_visualizing/data_preparation.rst (hunk 3 above)
Comment (banilo) [inline doc/manipulating_visualizing/data_preparation.rst]: looky looky

**Comment by banilo** [comment]
Thx for the pointers, just updated this one.

**Comment by AlexandreAbraham** [comment]
@lesteve can you do a pass on @banilo's PR?

File: doc/manipulating_visualizing/data_preparation.rst [hunk 8]
```diff
@@ -78,15 +89,15 @@ have to call low-level functions (see e.g. :mod:`nilearn.signal`,
...
 =========================================================
 
-This section describes how to use the :class:`NiftiMasker` class in
-more details than the previous description. :class:`NiftiMasker` is a
+This section details how to use the :class:`NiftiMasker` class.
+:class:`NiftiMasker` is a
 powerful tool to load images and extract voxel signals in the area
 defined by the mask. It is designed to apply some basic preprocessing
-steps by default with commonly used default parameters. But it is
+steps by default with commonly used parameters. But it is
```
Comment (lesteve) [inline doc/manipulating_visualizing/data_preparation.rst]: You are removing part of the meaning of this sentence as I understand it, i.e. that the default parameters are set to commonly used values. Maybe "with commonly used parameters as defaults"?

File: doc/manipulating_visualizing/data_preparation.rst [hunk 9]
```diff
@@ -98,11 +109,11 @@ Sometimes, some custom preprocessing of data is necessary. For instance
...
 a resting-state dataset with :func:`fetch_fetch_nyu_rest()
 <nilearn.datasets.fetch_nyu_rest>`, restrict it to 100 frames and
-build a brand new Nifti-like object to give it to the masker. Though it
-is possible, there is no need to save your data in a file to pass it to a
+build a brand new Nifti-like object to give it to the masker. Although
+possible, there is no need to save your data to a file to pass it to a
 :class:`NiftiMasker`. Simply use `nibabel
 <http://nipy.sourceforge.net/nibabel/>`_ to create a :ref:`Niimg <niimg>`
-in memory:
+dynamically in memory:
```
Comment (lesteve) [inline doc/manipulating_visualizing/data_preparation.rst]: I am not convinced dynamically adds anything here, especially given our target audience.

File: doc/manipulating_visualizing/data_preparation.rst [hunk 10]
```diff
@@ -112,10 +123,12 @@ in memory:
...
-In the basic tutorial, we showed how the masker could compute a mask
-automatically, and it has done a good job. But, on some datasets, the
-default algorithm performs poorly. This is why it is very important to
-**always look at how your data look like**.
+In this tutorial, we show how the masker object can compute a mask
+automatically for subsequent statistical analysis.
+On some datasets, the default algorithm may however perform poorly.
+This is why it is very important to
+**always look at your data** before and after feature
+engineering using masker objects.
```
Comment (lesteve) [inline doc/manipulating_visualizing/data_preparation.rst]: Just curious, does using a masker really qualify as feature engineering ?

File: doc/manipulating_visualizing/data_preparation.rst [hunk 11]
```diff
@@ -124,21 +137,25 @@ Computing the mask
...
     The full example described in this section can be found here:
     :doc:`plot_mask_computation.py <../auto_examples/manipulating_visualizing/plot_mask_computation>`.
-    This one can be relevant too:
+    It is also related to this example:
     :doc:`plot_nifti_simple.py <../auto_examples/plot_nifti_simple>`.
 
-If a mask is not given, :class:`NiftiMasker` will try to compute
-one. It is *very important* to take a look at the generated mask, to see if it
-is suitable for your data and adjust parameters if it is not. See the
+If a mask is not specified as an arugment,
```
Comment (lesteve) [inline doc/manipulating_visualizing/data_preparation.rst]: typo: arugment -> argument

File: doc/manipulating_visualizing/data_preparation.rst [hunk 12]
```diff
@@ -124,21 +137,25 @@ Computing the mask
...
-scratch. The Haxby dataset will be used since it provides a mask that we
-can use as a reference.
+As a first example, we will now automatically build a mask from a dataset.
+We will here use the Haxby dataset because it provides the original mask that
+we can compare the data-derived mask against.
 
-The first step is to generate a mask with default parameters and take
-a look at it.
+The first step is to generate a mask with default parameters and visualize it
+for eye inspection.
```
Comment (lesteve) [inline doc/manipulating_visualizing/data_preparation.rst]: "for" sounds weird, maybe "by" or maybe just stops the sentence as "visualize it".

File: doc/manipulating_visualizing/data_preparation.rst [hunk 13]
```diff
@@ -166,11 +185,11 @@ opening steps (*opening=10*) using the `mask_args` argument of the
 
 Looking at the :func:`nilearn.masking.compute_epi_mask` called by the
 :class:`NiftiMasker` object, we see two interesting parameters:
-*lower_cutoff* and *upper_cutoff*. These set the grey-values bounds in
-which the masking algorithm is going to try to find it's threshold (where
-0 is the minimum of the image, and 1 the maximum). Here we raise a lot
-the lower cutoff, and thus force the masking algorithm to select only
-voxels that are very light on the EPI image.
+*lower_cutoff* and *upper_cutoff*. These set the grey-value bounds in
+which the masking algorithm will search for it's threshold
```
Comment (lesteve) [inline doc/manipulating_visualizing/data_preparation.rst]: it's -> its

File: doc/manipulating_visualizing/data_preparation.rst [hunk 14]
```diff
@@ -199,21 +219,28 @@ Common data preparation steps: resampling, smoothing, filtering
 Resampling
 ..........
 
-:class:`NiftiMasker` and many similar classes enable resampling images.
+:class:`NiftiMasker` and many similar classes enable recasting of images
+       into different resolutions and constellations of brain voxel data.
```
Comment (lesteve) [inline doc/manipulating_visualizing/data_preparation.rst]: I find resampling clearer, but that may be me and the fact that I don't really understand what "constellations of brain voxel data" means.

File: doc/manipulating_visualizing/data_preparation.rst [hunk 15]
```diff
@@ -244,11 +271,14 @@ computes automatically the translation part of the affine.
...
 .........
 
-If smoothing the data prior to converting to voxel signals is required, it can
-be performed by :class:`NiftiMasker`. It is achieved by passing the full-width
-half maximum (in millimeter) along each axis in the parameter `smoothing_fwhm`.
+:class:`NiftiMasker` can further be used for local spatial filtering of
+the neuroimaging data to make the data more homogeneous and thus account
+for inter-individual differences in neuroanatomy.
+It is achieved by passing the full-width
+half maximum (FWHM; typically in millimeter scale)
```
Comment (lesteve) [inline doc/manipulating_visualizing/data_preparation.rst]: why "typically"? IIRC you have to pass it in millimeter to the function, right?

File: doc/manipulating_visualizing/data_preparation.rst [hunk 16]
```diff
@@ -244,11 +271,14 @@ computes automatically the translation part of the affine.
...
+along the x, y, and z image axes by specifying the `smoothing_fwhm` parameter.
 For an isotropic filtering, passing a scalar is also possible. The underlying
-function handles properly the tricky case of non-cubic voxels, by scaling the
+function handles properly this tricky case of non-cubic voxels by scaling the
```
Comment (lesteve) [inline doc/manipulating_visualizing/data_preparation.rst]: "the" sounds better to me.

File: doc/manipulating_visualizing/data_preparation.rst [hunk 17]
```diff
@@ -261,17 +291,27 @@ given widths appropriately.
 Temporal Filtering
 ..................
 
-All previous filters operate on images, before conversion to voxel signals.
+Rather than optimizing spatial properties of the neuroimaging data,
+the user typically wants to improve aspects of temporal data properties,
```
Comment (lesteve) [inline doc/manipulating_visualizing/data_preparation.rst]: "typically wants" -> "may want"

File: doc/manipulating_visualizing/data_preparation.rst [hunk 18]
```diff
@@ -261,17 +291,27 @@ given widths appropriately.
...
+before conversion to voxel signals.
 :class:`NiftiMasker` can also process voxel signals. Here are the possibilities:
 
 - Confound removal. Two ways of removing confounds are provided. Any linear
-  trend can be removed by activating the `detrend` option. It is not activated
-  by default in :class:`NiftiMasker` but is almost essential. More complex confounds can
+  trend can be removed by activating the `detrend` option.
+  This accounts for slowly (as opposed to abruptly or transiently) changes
```
Comment (lesteve) [inline doc/manipulating_visualizing/data_preparation.rst]: I think you need adjectives here, so:
* slowly -> slow
* abruptly -> abrupt
* transiently -> transient

Comment (lesteve) [inline doc/manipulating_visualizing/data_preparation.rst]: Also why did you remove the note about detrend=False by default although it is probably a good idea to set it to True in general ?

Comment (lesteve) [inline doc/manipulating_visualizing/data_preparation.rst]: Oops sorry scratch the last comment.

File: doc/manipulating_visualizing/data_preparation.rst [hunk 19]
```diff
@@ -310,51 +350,41 @@ Extraction of signals from regions:\  :class:`NiftiLabelsMasker`, :class:`NiftiM
...
-  represented (no weighting).
+  common label (e.g., anatomical brain region definitions as integers)
+  in the region definition array. The set of
+  regions is defined by a single 3D array, containing a voxel-wise
+  dictionary of label numbers that denote what
+  region a given voxel belongs to. This technique has a big advantage: the
+  required memory load is independent of the number of regions, allowing
+  for a large number of regions. On the other hand, there are
+  several disadvantages: regions cannot spatially overlap
+  (i.e., regions cannot overlap)
```
Comment (lesteve) [inline doc/manipulating_visualizing/data_preparation.rst]: "regions cannot spatially overlap" is quite redundant with "(i.e., regions cannot overlap)"

File: doc/manipulating_visualizing/data_preparation.rst [hunk 20]
```diff
@@ -310,51 +350,41 @@ Extraction of signals from regions:\  :class:`NiftiLabelsMasker`, :class:`NiftiM
...
+  and are represented in a binary present-nonpresent coding (no weighting).
 - maps: a single region is defined as the set of all the voxels that have a
   non-zero weight. A set of regions is thus defined by a set of 3D images (or a
-  single 4D image), one 3D image per region. Overlapping regions with weights
-  can be represented, but with a storage cost that scales linearly with the
-  number of regions. Handling a large number (thousands) of regions will prove
-  difficult with this representation.
+  single 4D image), one 3D image per region (as opposed to all regions in a
+  single 3D image such as for labels, cf. above).
+  While these the defined weighted regions can exhibit spatial
```
Comment (lesteve) [inline doc/manipulating_visualizing/data_preparation.rst]: "these the" pick one between the two

File: doc/manipulating_visualizing/data_preparation.rst [hunk 21]
```diff
@@ -365,9 +395,26 @@ instead of a set of maps as input.
 The `background_label` keyword of :class:`NiftiLabelsMasker` deserves
 some explanation. The voxels that correspond to the brain or a region
 of interest in an fMRI image do not fill the entire
-image. Consequently, in the labels image, there must be a label
-corresponding to "outside" the brain, for which no signal should be
-extracted.  By default, this label is set to zero in nilearn, and is
-referred to as "background". Should some non-zero value occur, it is
+image. Consequently, in the labels image, there must be a label value that
+corresponding to "outside" the brain (for which no signal should be
```
Comment (lesteve) [inline doc/manipulating_visualizing/data_preparation.rst]: either stick to "corresponding" or switch to "that corresponds"

File: doc/manipulating_visualizing/data_preparation.rst [hunk 22]
```diff
@@ -365,9 +395,26 @@ instead of a set of maps as input.
...
+extracted). By default, this label is set to zero in nilearn
+("background"). Should some non-zero value encoding be necessary, it is
```
Comment (lesteve) [inline doc/manipulating_visualizing/data_preparation.rst]: what was wrong with "and is referred as background"?

File: doc/manipulating_visualizing/data_preparation.rst (hunk 8 above)
Comment (banilo) [inline doc/manipulating_visualizing/data_preparation.rst]: Ok.

File: doc/manipulating_visualizing/data_preparation.rst (hunk 10 above)
Comment (banilo) [inline doc/manipulating_visualizing/data_preparation.rst]: yes. Feature Agglomeration woould be plain Masking, but Feature Engineering implies something smarter than just selecting.

**Comment by banilo** [comment]
:octocat: Nice PR, merge it now!

**Comment by lesteve** [comment]
Merging, thanks a lot!
